benches/benches/bevy_ecs/change_detection.rs

```diff
@@ -109,19 +125,19 @@ fn all_changed_detection_generic<T: Component + Default + BenchModify>(
     group.bench_function(
         format!("{}_entities_{}", entity_count, std::any::type_name::<T>()),
         |bencher| {
-            bencher.iter_batched(
+            bencher.iter_batched_ref(
                 || {
                     let mut world = setup::<T>(entity_count);
                     world.clear_trackers();
                     let mut query = world.query::<&mut T>();
                     for mut component in query.iter_mut(&mut world) {
                         black_box(component.bench_modify());
                     }
-                    world
+                    let query = world.query_filtered::<Entity, Changed<T>>();
```

Comment:
There's a lot of code duplicated between these benchmarks that it's important not to let desync. Can we pull these out into a function that's generic over the query type?

Reply:
fine!